### PR TITLE
feat(builder): Addition of prettier for aligned of development styles

### PR DESCRIPTION
**Pull request by andrewhooker2**
### Background

Resolves: [Implement Prettier for Code Formatting #7628](https://github.com/Significant-Gravitas/AutoGPT/issues/7628)

We currently do not have a formatter in action for our "Builder". This PR will help move PR's across the line faster and align our development styles.

### Changes 🏗️

- Addition of `.prettierignore`
      - Ignoring files that are not necessary for format
- Addition of `.prettierrc.json`
      - Configuration of the prettier implementation 
- Addition of a job in the `autogpt-builder-ci.yml` file
      - Addition of a job that will run the script of format leveraging prettier. 
-    The last change was to run "format" script on the repo.

### PR Quality Scorecard ✨
- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [x] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
Enhancement, Other

___

### **Description**
- Added Prettier configuration and ignore files to standardize code formatting.
- Applied Prettier formatting to multiple files across the repository.
- Added a job in the CI workflow to run Prettier formatting.
- Updated `package.json` to include Prettier as a dependency and added a format script.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Formatting</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Apply Prettier formatting to monitor page component.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/app/monitor/page.tsx

<li>Applied Prettier formatting to the entire file.<br> <li> Adjusted import statements to use single quotes.<br> <li> Reformatted function calls and object properties for better <br>readability.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7629/files#diff-c0277d8d99528e90bb8cb9c6dccd0e76576618553ef642d5cd55c584e692973d">+564/-383</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Flow.tsx</strong><dd><code>Apply Prettier formatting to Flow component.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/components/Flow.tsx

<li>Applied Prettier formatting to the entire file.<br> <li> Adjusted import statements to use single quotes.<br> <li> Reformatted function calls and object properties for better <br>readability.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7629/files#diff-31d9430263189bf5d64454c82b2151fe2b4198ea7c8b3beb36a99563c5dcc00d">+311/-203</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>NodeInputField.tsx</strong><dd><code>Apply Prettier formatting to NodeInputField component.</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/components/NodeInputField.tsx

<li>Applied Prettier formatting to the entire file.<br> <li> Adjusted import statements to use single quotes.<br> <li> Reformatted function calls and object properties for better <br>readability.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7629/files#diff-f1514eb3655323ef33ef6827ca06951ea8fece147a903ccf45aa35ec2e7fa6d3">+304/-229</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>dropdown-menu.tsx</strong><dd><code>Apply Prettier formatting to dropdown menu component.</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/components/ui/dropdown-menu.tsx

<li>Applied Prettier formatting to the entire file.<br> <li> Adjusted import statements to use single quotes.<br> <li> Reformatted function calls and object properties for better <br>readability.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7629/files#diff-ff0d2b0f1fcb99c9e65edf40ba230a32cd28f1f1b93a0b7323b0721d3da16e91">+50/-47</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>CustomNode.tsx</strong><dd><code>Apply Prettier formatting to CustomNode component.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/components/CustomNode.tsx

<li>Applied Prettier formatting to the entire file.<br> <li> Adjusted import statements to use single quotes.<br> <li> Reformatted function calls and object properties for better <br>readability.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7629/files#diff-ce37ba33e874cc42255fda0cd2a16758a94a6e68606d513b7018c1cc68193146">+106/-54</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>client.ts</strong><dd><code>Apply Prettier formatting to AutoGPT server API client.</code>&nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/lib/autogpt-server-api/client.ts

<li>Applied Prettier formatting to the entire file.<br> <li> Adjusted import statements to use single quotes.<br> <li> Reformatted function calls and object properties for better <br>readability.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7629/files#diff-234c98a3eb2abb20de2a7e2c6ccd167015ef7a5c11844ef0300af364cf276b3c">+81/-58</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Apply Prettier formatting to AutoGPT server API index.</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/lib/autogpt-server-api/index.ts

<li>Applied Prettier formatting to the entire file.<br> <li> Adjusted import statements to use single quotes.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7629/files#diff-8ea2588cebd2490525462daea2bf11c69127967484b14bd0d813cce42b286352">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>components.json</strong><dd><code>Apply Prettier formatting to components.json.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/components.json

- Applied Prettier formatting to the JSON file.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7629/files#diff-4ed51a810655ea652210f6e98354c55f7ce0052cb4ad16695550eb5269ffb05c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: rnd/autogpt_builder/.prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (Pwuts) [inline rnd/autogpt_builder/.prettierrc.json]: Let's work with Prettier's defaults for now. No need for this file.

File: rnd/autogpt_builder/.prettierignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+node_modules
+.next
+build
+public
```
Comment (Pwuts) [inline rnd/autogpt_builder/.prettierignore]: Missing newline

**Review by Pwuts — CHANGES_REQUESTED** [review]
This is a very welcome DX addition! Could you also add a formatting step to the repo's pre-commit config?

File: .github/workflows/autogpt-builder-ci.yml [hunk 3]
```diff
@@ -17,6 +17,24 @@ defaults:
...
+        with:
+          node-version: '21'
+
+      - name: Install dependencies
+        run: |
+          npm install
+
+      - name: Run format
+        run: |
+          npm run format
```
Comment (ntindle) [inline .github/workflows/autogpt-builder-ci.yml]: Run with —check https://prettier.io/docs/en/cli.html#--check

File: rnd/autogpt_builder/.prettierrc.json (hunk 1 above)
Comment (andrewhooker2) [reply rnd/autogpt_builder/.prettierrc.json]: Sounds good, I'll keep it to the defaults suggested on their site:
[Configuration File](https://prettier.io/docs/en/configuration.html)

File: rnd/autogpt_builder/.prettierignore (hunk 2 above)
Comment (andrewhooker2) [reply rnd/autogpt_builder/.prettierignore]: Thanks for the catch I'll add it in

File: .github/workflows/autogpt-builder-ci.yml (hunk 3 above)
Comment (andrewhooker2) [reply .github/workflows/autogpt-builder-ci.yml]: Great call out here @ntindle, 

I have updated the ci to no longer add an additional job but instead run the check before linting as linting is generally more costly. 

If prettier was not ran before the push the job will fail. 

I'm not sure if we wanted an error message to be echoed here so I left it out. Please let me know if it is requested.

File: rnd/autogpt_builder/.prettierrc.json (hunk 1 above)
Comment (Pwuts) [reply rnd/autogpt_builder/.prettierrc.json]: Those are not the defaults. Defaults are listed [here](https://prettier.io/docs/en/options). Please just delete the file.

Comment (andrewhooker2) [reply rnd/autogpt_builder/.prettierrc.json]: Can do. I have removed the file.

**Comment by Swiftyos** [comment]
Would you mind only committing the changes to add prettier and not the actually formatting changes. 

We can run the formatting changes straight after reviewing and merging these changes.

We appreciate the work your doing, Thank you.

**Comment by andrewhooker2** [comment]
@Swiftyos, with the addition of the check to the CI. That would not be possible. 
This is because the pipeline will fail if prettier was not ran before push.

**Comment by Swiftyos** [comment]
Yes I can merge it even with that check failing. Then I'll fix the CI by formatting it :)

**Comment by andrewhooker2** [comment]
Perfect thank you @Swiftyos, I've shifted the focus to be the files changed only.

**Comment by Swiftyos** [comment]
Perfect thank you! We'll coordinate merging this in with some of or other bigger frontend PR's and get it ran.

Really appreciate this 🙏

**Review by Swiftyos — APPROVED** [review]

**Pull request merged**